src/test/java/de/tum/in/www1/artemis/ExampleSubmissionIntegrationTest.java

```diff
@@ -243,4 +250,73 @@ public void createExampleTextAssessment_wrongExerciseId() throws Exception {
         assertThat(exampleSubmissionRepo.findBySubmissionId(randomId)).isEmpty();
     }
 
+    private ExampleSubmission importExampleSubmission(HttpStatus expectedStatus, Long submissionId, Long exerciseId) throws Exception {
+        return request.postWithResponseBody("/api/exercises/" + exerciseId + "/example-submissions/import", submissionId, ExampleSubmission.class, expectedStatus);
+    }
+
+    @Test
+    @WithMockUser(value = "instructor1", roles = "INSTRUCTOR")
+    public void importExampleSubmissionWithTextSubmission() throws Exception {
+        Submission submission = ModelFactory.generateTextSubmission("submissionText", Language.ENGLISH, true);
+        submission = database.saveTextSubmission(textExercise, (TextSubmission) submission, "student1");
+
+        TextBlock textBlock = new TextBlock();
+        textBlock.setCluster(null);
+        textBlock.setAddedDistance(0);
+        textBlock.setStartIndex(0);
+        textBlock.setEndIndex(14);
+        database.addAndSaveTextBlocksToTextSubmission(Set.of(textBlock), (TextSubmission) submission);
+
+        database.addResultToSubmission(submission, AssessmentType.MANUAL);
+
+        // add one feedback for the created text block
+        List<TextBlock> textBlocks = new ArrayList<>(((TextSubmission) submission).getBlocks());
+        Feedback feedback = new Feedback();
+        if (!textBlocks.isEmpty()) {
```

Comment:
Should we fail the test if the textBlocks are empty? Since at this point, we kinda expect that there are some since we added them. If there are none, this should be an exceptional behaviour that must be fixed, shouldn't it?

Reply:
hmm did not think about it, yeah could be.